app/lib/submission_builder/ty2024/states/nj/documents/nj1040.rb

```diff
@@ -26,21 +30,22 @@ def document
                     case status
                     when :married_filing_separately
                       xml.MarriedCuPartFilingSeparate do
-                        xml.SpouseSSN intake.spouse_ssn
+                        xml.SpouseSSN intake.spouse.ssn
                         xml.SpouseName do
-                          xml.FirstName sanitize_for_xml(intake.spouse_first_name)
-                          xml.MiddleInitial sanitize_for_xml(intake.spouse_middle_initial) if intake.spouse_middle_initial.present?
-                          xml.LastName sanitize_for_xml(intake.spouse_last_name)
-                          xml.NameSuffix intake.spouse_suffix if intake.spouse_suffix.present?
+                          xml.FirstName sanitize_for_xml(intake.spouse.first_name)
+                          xml.MiddleInitial sanitize_for_xml(intake.spouse.middle_initial) if intake.spouse.middle_initial.present?
+                          xml.LastName sanitize_for_xml(intake.spouse.last_name)
+                          xml.NameSuffix intake.spouse.suffix if intake.spouse.suffix.present?
                         end
                       end
                     when :qualifying_widow
                       yod = Date.parse(@submission.data_source.direct_file_data.spouse_date_of_death)&.strftime("%Y")
                       xml.QualWidOrWider do
                         xml.QualWidOrWiderSurvCuPartner 'X'
-                        if yod == Time.now.year - 1
+                        case yod
+                        when '2024'
                           xml.LastYear 'X'
-                        elsif yod == Time.now.year - 2
+                        when '2023'
```

Comment:
[dust] should we be worried about hardcoding these years from a future-proofing perspective?